### PR TITLE
feat: IPRPC over IBC: Part 5 - CNS-967: Cover pending IBC IPRPC fund costs

### PR DESCRIPTION
**Pull request by oren-lava**
In this PR, I implemented the `cover-ibc-iprpc-fund-cost` TX. This TX allows a lava user to send `ulava` to cover the minimum IPRPC funding cost for a specific PendingIbcIprpcFund (by index).

This TX automatically takes the required cost, funds the IPRPC pools, deletes the PendingIbcIprpcFund object and sends leftovers to the community pool. If the TX is issued by the gov module (using a proposal), the PendingIbcIprpcFund is applied without paying the minimum IPRPC cost.

### CODE REVIEW

File: proto/lavanet/lava/rewards/tx.proto [hunk 1]
```diff
@@ -37,4 +38,12 @@ message MsgFundIprpc {
 message MsgFundIprpcResponse {
 }
 
+message MsgCoverIbcIprpcFundCost {
+    string creator = 1;
+    uint64 index = 2; // PendingIbcIprpcFund index
```
Comment (Yaroms) [inline proto/lavanet/lava/rewards/tx.proto:43]: maybe we should let this be indices?

Comment (Yaroms) [reply proto/lavanet/lava/rewards/tx.proto:43]: or at least in the CLI

File: x/rewards/keeper/ibc_iprpc.go [hunk 2]
```diff
@@ -289,3 +290,61 @@ func (k Keeper) CalcPendingIbcIprpcFundMinCost(ctx sdk.Context, pendingIbcIprpcF
...
+	piif, found := k.GetPendingIbcIprpcFund(ctx, index)
+	if !found {
+		return zeroCoin, utils.LavaFormatWarning("PendingIbcIprpcFund not found", types.ErrCoverIbcIprpcFundCostFailed,
+			utils.LogAttr("creator", creator),
+			utils.LogAttr("index", index),
+		)
+	}
+
+	// sanity check: PendingIbcIprpcFund is not expired
+	if piif.IsExpired(ctx) {
```
Comment (Yaroms) [inline x/rewards/keeper/ibc_iprpc.go:316]: if its expired it shouldnt be here

File: x/rewards/keeper/iprpc.go [hunk 3]
```diff
@@ -17,49 +17,56 @@ func (k Keeper) FundIprpc(ctx sdk.Context, creator string, duration uint64, fund
...
-
-	// check creator has enough balance
-	addr, err := sdk.AccAddressFromBech32(creator)
-	if err != nil {
-		return utils.LavaFormatWarning("invalid creator address", types.ErrFundIprpc)
-	}
+	// calculate total funds to transfer to the IPRPC pool (input fund is monthly fund)
+	totalFunds := fund.MulInt(math.NewIntFromUint64(duration))
+
+	// if the fund TX originates from the gov module (keeper's authority field) it's not paying the minimum IPRPC cost
```
Comment (Yaroms) [inline x/rewards/keeper/iprpc.go:23]: update the comment also for types.IbcIprpcReceiver

Comment (oren-lava) [reply x/rewards/keeper/iprpc.go]: outdated. Resolving.

**Pull request merged**